src/user-event/__tests__/select-options.js

```diff
@@ -0,0 +1,79 @@
+import * as userEvent from '..'
+import {setupSelect} from './helpers/utils'
+
+test('fires correct events', async () => {
+  const {select, options, getEventCalls} = setupSelect()
+  await userEvent.selectOptions(select, '1')
+  expect(getEventCalls()).toMatchInlineSnapshot(`
+    Events fired on: select[name="select"][value="1"]
+
+    mouseover: Left (0)
+    mousemove: Left (0)
+    mousedown: Left (0)
+    focus
+    mouseup: Left (0)
+    click: Left (0)
+    mouseover: Left (0) (bubbled from option[value="1"])
+    mousemove: Left (0) (bubbled from option[value="1"])
+    mousedown: Left (0) (bubbled from option[value="1"])
+    mouseup: Left (0) (bubbled from option[value="1"])
+    click: Left (0) (bubbled from option[value="1"])
+    change
+  `)
+  const [o1, o2, o3] = options
+  expect(o1.selected).toBe(true)
+  expect(o2.selected).toBe(false)
+  expect(o3.selected).toBe(false)
+})
+
+test('fires correct events on multi-selects', async () => {
+  const {select, options, getEventCalls} = setupSelect({multiple: true})
+  await userEvent.selectOptions(select, ['1', '3'])
+  expect(getEventCalls()).toMatchInlineSnapshot(`
+    Events fired on: select[name="select"][value=["1","3"]]
+
+    mouseover: Left (0)
+    mousemove: Left (0)
+    mousedown: Left (0)
+    focus
+    mouseup: Left (0)
+    click: Left (0)
+    mouseover: Left (0) (bubbled from option[value="1"])
+    mousemove: Left (0) (bubbled from option[value="1"])
+    mousedown: Left (0) (bubbled from option[value="1"])
+    mouseup: Left (0) (bubbled from option[value="1"])
+    click: Left (0) (bubbled from option[value="1"])
+    change
+    mouseover: Left (0) (bubbled from option[value="3"])
+    mousemove: Left (0) (bubbled from option[value="3"])
+    mousedown: Left (0) (bubbled from option[value="3"])
+    mouseup: Left (0) (bubbled from option[value="3"])
+    click: Left (0) (bubbled from option[value="3"])
+    change
+  `)
+  const [o1, o2, o3] = options
+  expect(o1.selected).toBe(true)
+  expect(o2.selected).toBe(false)
+  expect(o3.selected).toBe(true)
+})
+
+test('sets the selected prop on the selected OPTION using nodes', async () => {
```

Comment:
Is "using nodes" important here? I can't see how "nodes" are used in this test. Is this an implementation detail?

Reply:
The feature is that you can pass OPTION nodes to the function and it will select those options. That's an exposed feature so not an implementation detail.

Reply:
```suggestion
test('sets the selected prop on the selected OPTION using option html elements', async () => {
```
?